File: examples/openai-langchain.py
"""
openai-langchain.py — ChatOpenAI through the Tessera proxy.

Usage:

    pip install tessera-langchain langchain-openai
    export OPENAI_API_KEY=sk-...
    export TESSERA_API_KEY=tsr_...
    python openai-langchain.py
"""

from __future__ import annotations

import os

from langchain_core.messages import HumanMessage, SystemMessage
from langchain_openai import ChatOpenAI

from tessera_langchain import tessera_openai_config


def main() -> None:
    openai_key = os.environ["OPENAI_API_KEY"]
    tessera_key = os.environ["TESSERA_API_KEY"]

    llm = ChatOpenAI(
        model="gpt-4o",
        openai_api_key=openai_key,
        **tessera_openai_config(api_key=tessera_key),
    )

    response = llm.invoke(
        [
            SystemMessage(
                content=(
                    "You are a senior platform engineer reviewing infrastructure "
                    "decisions. Answer in 3 concise bullets."
                )
            ),
            HumanMessage(
                content=(
                    "Compare a service mesh sidecar approach vs an eBPF-based "
                    "approach for east-west traffic policy enforcement."
                )
            ),
        ]
    )

    print(response.content)


if __name__ == "__main__":
    main()

File: examples/openai-langchain.ts
/**
 * openai-langchain.ts — ChatOpenAI through the Tessera proxy.
 *
 * Usage:
 *
 *   npm install @tessera-llm/langchain @langchain/openai @langchain/core
 *   export OPENAI_API_KEY=sk-...
 *   export TESSERA_API_KEY=tsr_...
 *   npx tsx openai-langchain.ts
 */

import { ChatOpenAI } from "@langchain/openai";
import { HumanMessage, SystemMessage } from "@langchain/core/messages";

import { tesseraOpenAIConfig } from "@tessera-llm/langchain";

async function main(): Promise<void> {
  const openaiKey = process.env.OPENAI_API_KEY;
  const tesseraKey = process.env.TESSERA_API_KEY;
  if (!openaiKey || !tesseraKey) {
    throw new Error("Set OPENAI_API_KEY and TESSERA_API_KEY before running.");
  }

  const llm = new ChatOpenAI({
    model: "gpt-4o",
    apiKey: openaiKey,
    ...tesseraOpenAIConfig({ apiKey: tesseraKey }),
  });

  const response = await llm.invoke([
    new SystemMessage(
      "You are a senior platform engineer reviewing infrastructure decisions. " +
        "Answer in 3 concise bullets."
    ),
    new HumanMessage(
      "Compare a service mesh sidecar approach vs an eBPF-based approach " +
        "for east-west traffic policy enforcement."
    ),
  ]);

  console.log(response.content);
}

main().catch((err) => {
  console.error(err);
  process.exitCode = 1;
});
